pkg/storer/internal/upload/uploadstore.go

```diff
@@ -848,15 +827,31 @@ func DeleteTag(st storage.Store, tagID uint64) error {
 	return nil
 }
 
-func IterateAll(st storage.Store, iterateFn func(addr swarm.Address, isSynced bool) (bool, error)) error {
+func IterateAll(st storage.Store, iterateFn func(item storage.Item) (bool, error)) error {
 	return st.Iterate(
 		storage.Query{
 			Factory: func() storage.Item { return new(uploadItem) },
 		},
 		func(r storage.Result) (bool, error) {
-			address := swarm.NewAddress([]byte(r.ID[:32]))
-			synced := r.Entry.(*uploadItem).Synced != 0
-			return iterateFn(address, synced)
+			ui := r.Entry.(*uploadItem)
+			ui.IdFunc = func() string {
```

Comment:
what is the purpose of this?

Reply:
when uploadItems are marshalled, the address and batchID (which make up the ID) are not stored. 
It was assumed that the user would always know these properties when trying to get the item from the store. 
```
ui := &uploadItem {
   Address: addr,
   BatchID: id,
}
store.Get(ui)
```
So there was no need to save the address and batch to storage.

But this is a problem when iterating the store. 
Since the address and batchID were not stored, the resulting upload items don't have a unique ID.
```
iterate(func (r Result) {
  ui := r.Entry.(*uploadItem)
  println(ui.ID() == "/") // true (since address and batchID are empty, ID() == "/" for all items
  
  // Luckily, the ID is stored in the `Result`. I introduced IdFunc here as a workaround
  ui.IdFunc = func() { return r.ID }
}
```

Reply:
got it, thanks.